tests/test_main.py

```diff
@@ -649,3 +649,31 @@ class Config:
     m = MyModel(extra_key='extra')
     assert m.dict() == {'a': None, 'extra_key': 'extra'}
     assert m.dict(by_alias=True) == {'alias_a': None, 'extra_key': 'extra'}
+
+
+def test_alias_provider():
+    def to_camel(string: str):
+        return ''.join(x.capitalize() for x in string.split('_'))
+
+    class MyModel(BaseModel):
+        a: List[str] = None
+        foo_bar: str
+
+        class Config:
+            alias_provider = to_camel
+
+    data = {'A': ['foo', 'bar'], 'FooBar': 'foobar'}
+    v = MyModel(**data)
+    assert v.a == ['foo', 'bar']
+    assert v.foo_bar == 'foobar'
+    assert v.dict(by_alias=True) == data
+
+    with pytest.raises(TypeError) as e:
+
+        class MyModel(BaseModel):
+            bar: Any
+
+            class Config:
+                alias_provider = str.encode
```

Comment:
this will cause 
```bash
flake8 pydantic/ tests/
tests/test_main.py:676:17: E731 do not assign a lambda expression, use a def
make: *** [lint] Error 1
```

Reply:
can do something like:
```python
class Config:
    @staticmethod
    def alias_generator(string: str):
        return b'not a string'
```
or 
```python
def return_bytes(string):
    return b'not a string'
...
    class Config:
        alias_generator = return_bytes

Reply:
yes, second options. Or since it's a test you could also do `# noqa`